demos/ws-bootable-jar-example/jaxws/src/main/java/org/jboss/jaxws/Endpoint.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * JBoss, Home of Professional Open Source.
```

Comment:
Could you please run the following commands?

```
 mvn spotless:apply -pl demos
 ```
 
 and  
 ```
 mvn process-sources
 ```
 
 The former will fix the headers for the new files,  the latter will forma the sources.

Reply:
For me spotless does not alter any files.
Because demos is not declared in intersmash/pom.xml
I added spotless plugin references in demoes/ws-bootable-jar-example/pom.xml
and updated the reference to license-header.
I ran cmd "mvn spotless:apply -pl ws-bootable-jar-example" from the demos
dir.  No file licenses were updated.  I tried removing the existing license
text from the files and ran spotless again.  No text was inserted.
I can manually update the license text.  Is that sufficient?

No sources were altered with mvn process-sources.

Reply:
Here I am @rsearls - thanks for bringing this up. For sure this spotless thing will have to be documented.

Anyway, I'd propose to add a profile to the intersmash parent POM, i.e. namely the `demo` profile, which can be activated by passing `-Pdemo` to `mvn`.
This profile definition should just add the `demos` module to the project modules, i.e. something like:
```
<profiles>
        <profile>
            <id>demo</id>
            <modules>
                <module>demos</module>
            </modules>
        </profile>
    </profiles>
```

This way, by enabling such profile, we know the `demos` module and its submodules will be processed by Maven, i.e.:

* `mvn clean install -DskipTests`  will build as usual
* `mvn clean install -DskipTests -Pdemo`  will build the `demos` module as well.

This would similarly allow to run demo tests (`mvn clean test... -Pdemo`) selectively. 
Eventually, this would work for any other Maven plugin, including spotless. I've experimented a bit and I've got some changed files, see https://github.com/rsearls/intersmash/compare/jaxws-bootable-jar-test...fabiobrz:intersmash:jaxws-bootable-jar-test.fabiobrz.demo?expand=1 by doing:

```
$ mvn spotless:apply -pl demos/spotless-maven-plugin -Pdemo
...
$ mvn spotless:apply -pl demos/ws-bootable-jar-example -Pdemo
...
$ mvn spotless:apply -pl demos/ws-bootable-jar-example/jaxws -Pdemo
```

and finally:
```
$ mvn process-sources -Pdemo
```

This seems to me a good way to wrap up the demo that you provided and to make it easily usable, also as a starting point for adding more.
WDYT?
You can see a preview of the changes I've done myself in order to experiment a bit here: https://github.com/fabiobrz/intersmash/commit/1d31f226b9e65fbcbb7330f3f966335dbdb49ce7

Reply:
done